### PR TITLE
Typing fixes in config.py

### PR DESCRIPTION
**Pull request by ProducerMatt**

### CODE REVIEW

**Review by Aprillion — APPROVED** [review]

File: config.py [hunk 1]
```diff
@@ -32,7 +32,7 @@ def getenv(env_var: str, default: T) -> Union[str, T]:...
 @overload
 def getenv(env_var: str) -> str:...
 
-def getenv(env_var: str, default: T = NOT_PROVIDED) -> Union[str, T]:
+def getenv(env_var: str, default = NOT_PROVIDED) -> str:
```
Comment (MatthewBaggins) [inline config.py:35]: This is inconsistent/not how `@overload` works. `@overload`ed function signatures narrow down the return type, given a particular choice of argument types. Annotations on the function implementation itself (without `@overload`) should be a strict superset of the `@overload`ed ones. Here, `str` is a subset of `Union[str, T]`.

Comment (Aprillion) [reply config.py:35]: could we make 1 function always return 1 type (`str` in this case) and use a different funcition name for the other types? or what is the benefit of the overload / generics type?

Comment (ProducerMatt) [reply config.py:35]: @MatthewBaggins originally it was `default: T = NOT_PROVIDED`, but this raised typing errors saying `NOT_PROVIDED` was type str, not T. This change fixes that error, with no other errors raised anywhere else in the file. What do you think needs to be different?

Comment (ProducerMatt) [reply config.py:35]: @MatthewBaggins I'm going to go ahead and merge, let me know if you have an alternative way to do it that doesn't raise errors.

**Pull request merged**